### PR TITLE
Paste here

### PR DESCRIPTION
**Pull request by andyjda**
See [issue 994](https://github.com/Alexander-Miller/treemacs/issues/994)

### CODE REVIEW

File: src/elisp/treemacs-mouse-interface.el [hunk 1]
```diff
@@ -244,17 +244,17 @@ and ignore any prefix argument."
 (defun treemacs--tag-noselect (btn)
   "Return list of tag source buffer and position for BTN for future display."
   (cl-flet ((xref-definition
-             (identifier)
-             "Return the first definition of string IDENTIFIER."
-             (car (xref-backend-definitions (xref-find-backend) identifier)))
+              (identifier)
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:247]: Indentation shouldn't change here. The latest commit adds a .dir-locals file that should preclude unwanted diffs like this.

File: src/elisp/treemacs-mouse-interface.el [hunk 2]
```diff
@@ -277,9 +277,27 @@ and ignore any prefix argument."
                (list (xref-item-buffer xref) (xref-item-position xref)))))
           ('issue-warning
            (treemacs-log-failure "Tag '%s' is located in a buffer that does not exist."
-                          (propertize (treemacs-with-button-buffer btn (treemacs--get-label-of btn)) 'face 'treemacs-tags-face)))
+             (propertize (treemacs-with-button-buffer btn (treemacs--get-label-of btn)) 'face 'treemacs-tags-face)))
           (_ (error "[Treemacs] '%s' is an invalid value for treemacs-goto-tag-strategy" treemacs-goto-tag-strategy)))))))
 
+(defun treemacs--paste-point-to-minibuffer ()
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:283]: This command is public, so its name needs to have only one dash.

It is also not strictly part of the mouse interface, but is a more general utility, so it should be part of treemacs-interface.el. 

The name should also change to paste-path-at-point-to-minibuffer so it's clear what exactly it does.

File: src/elisp/treemacs-mouse-interface.el [hunk 3]
```diff
@@ -277,9 +277,27 @@ and ignore any prefix argument."
...
+  "Paste the path at point into the minibuffer.
+This is used by the \"Paste here\" button,
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:285]: Doc should specify it's the button from the mouse menu.

File: src/elisp/treemacs-mouse-interface.el [hunk 4]
```diff
@@ -277,9 +277,27 @@ and ignore any prefix argument."
...
+which assumes that we are running `treemacs--copy-or-move',
+so that pasting this path into the minibuffer allows us to copy/move
+the previously-selected file into the path at point."
+  (interactive)
+  (let* ((path-at-point (treemacs--prop-at-point :path))
+         (path
+          (if (file-directory-p path-at-point)
+              path-at-point
+            (file-name-directory path-at-point))))
+    (switch-to-minibuffer)
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:295]: That's for interactive use only. Use `(select-window (minibuffer-window))`

File: src/elisp/treemacs-mouse-interface.el [hunk 5]
```diff
@@ -277,9 +277,27 @@ and ignore any prefix argument."
...
+    (mark-whole-buffer)
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:296]: Same here, use `(delete-region (point-min) (point-max))` or `delete-minibuffer-contents`.

File: src/elisp/treemacs-mouse-interface.el [hunk 6]
```diff
@@ -277,9 +277,27 @@ and ignore any prefix argument."
...
+    (delete-region (region-beginning) (region-end))
+    (insert path))
+  (message "copied from treemacs"))
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:299]: `treemacs-log`

File: src/elisp/treemacs-mouse-interface.el [hunk 7]
```diff
@@ -300,7 +318,10 @@ and ignore any prefix argument."
             (menu
              (easy-menu-create-menu
               nil
-              `(("New"
+              `(["Paste here"
+                 treemacs--paste-point-to-minibuffer
+                 :visible ,(string-match-p "\\(\\(Move\\)\\|\\(Copy\\)\\) to: " (or (minibuffer-prompt) ""))]
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:323]: `(rx (or "Copy" "Move") " to:")`

File: src/elisp/treemacs-mouse-interface.el [hunk 8]
```diff
@@ -300,7 +318,10 @@ and ignore any prefix argument."
...
+                ("New"
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-mouse-interface.el:324]: Finally at the end the commits need to be squashed and the commit message prefixed with `[Interface]`.

**Review by Alexander-Miller — CHANGES_REQUESTED** [review]

**Comment by andyjda** [comment]
applied these changes; let me know if I missed something or if you need anything else from me :)

**Comment by Alexander-Miller** [comment]
Almost there, only some nitpicks left.

**Comment by andyjda** [comment]
Do you need me to make any additional changes? I thought I fixed all issues pointed out in your comments; let me know if I'm missing something or if there are additional edits needed

**Review by Alexander-Miller — CHANGES_REQUESTED** [review]

**Comment by Alexander-Miller** [comment]
Looks like I forgot to submit my review. You should see the new comments now.

**Comment by andyjda** [comment]
Thanks for the review, just pushed the changes :)
Let me know if there's anything else I'm missing

**Comment by andyjda** [comment]
@Alexander-Miller any update on this?

**Comment by Alexander-Miller** [comment]
> any update on this?

Apparently the update is that I sometimes forget to press the "Review Changes" button so my comments never show up. It doesn't help that github shows me that changes have been requested already. My bad :(

c

Anyway, no more delays now, I'll take care about my findings myself and merge in a day or two.

**Review by Alexander-Miller — APPROVED** [review]

**Comment by Alexander-Miller** [comment]
All done now. I've moved a few things around, let me know if anything is missing.

**Comment by andyjda** [comment]
looks perfect! Thank you

**Pull request merged**